prepare_source_tarball.sh

```diff
@@ -35,10 +35,40 @@ ORIG_DIR="$(pwd)"
 	git -C ${SRC_DIR} submodule update --init supportlibs/rapidjson 
 }
 
+[ "$(ls "${SRC_DIR}/supportlibs/libsam3" | wc -l)" -lt "5" ] &&
+{
+	git -C ${SRC_DIR} submodule update --init supportlibs/libsam3
+}
+
+[ "$(ls "${SRC_DIR}/supportlibs/cmark" | wc -l)" -lt "5" ] &&
+{
+	git -C ${SRC_DIR} submodule update --init supportlibs/cmark
+}
+
+[ "$(ls "${SRC_DIR}/supportlibs/jni.hpp" | wc -l)" -lt "5" ] &&
+{
+	git -C ${SRC_DIR} submodule update --init supportlibs/jni.hpp
+}
+
+[ "$(ls "${SRC_DIR}/libbitdht" | wc -l)" -lt "1" ] &&
+{
+	git -C ${SRC_DIR} submodule update --init --remote --force libbitdht
+}
+
+[ "$(ls "${SRC_DIR}/libretroshare" | wc -l)" -lt "1" ] &&
+{
+	git -C ${SRC_DIR} submodule update --init --remote --force libretroshare
+}
+
+[ "$(ls "${SRC_DIR}/openpgpsdk" | wc -l)" -lt "1" ] &&
+{
+	git -C ${SRC_DIR} submodule update --init --remote --force openpgpsdk
+}
+
 cd "${WORK_DIR}"
 rsync -a --delete \
 	--exclude='.git' \
-	--filter=':- build_scripts/OBS/.gitignore' \
```

Comment:
why removing the filter? and not just adding `--exclude='.osc/*'` in a new line?

Reply:
From a clean checkout, there didn't appear to be anything except the .osc folder that needed excluded. This could theoretically include extras in the event you're running it from a folder you've compiled locally in.

The issue I ran into is that this filter doesn't just include the .gitignore inside the 'OBS' folder, but EVERY .gitignore in the submodules as well. This resulted in some Makefiles being excluded from the tarball, which caused compilation to fail.

Reply:
That it take in accounts all `.gitignores` from all submodules is really unexpected, I have accepted your change for now it seems it will be good, in future if we see some garbage get included in the tarball we will see how to fix it ;)